tests/boundaries/HardBoundary.test.cpp

```diff
@@ -0,0 +1,175 @@
+#include <boundaries/HardBoundary.h>
+#include <gtest/gtest.h>
+#include <physicsCalculator/LJCalculator.h>
+
+// Test the ghost boundary in x direction for a single near particle
+TEST(HardBoundary, ParticleReflectCorrectly) {
+    // Initialize the list of particles
+    std::vector<Particle> particles = {
+        Particle({ -1.0, 0.0, 2.0 }, { 2.0, 1.0, 1.0 }, 1.0),
+        Particle({ 1.0, 12.0, 2.0 }, { 1.0, -3.0, -2.0 }, 1.0),
+        Particle({ 3.0, -2.0, 13.0 }, { 2.0, 2.0, -1.0 }, 1.0),
+        Particle({ 2.0, 5.0, -2.0 }, { 1.0, 3.0, -2.0 }, 1.0),
+    };
```

Comment:
In general, I would recommend to create one of the container you have implemented, add particles to it and carry out the tests.